res/decoder-ring/datatypes.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+(function(global) {
+    class Optional {
```

Comment:
There's 3 different design patterns in these 3 lines :(

We don't need `use strict` in 2020, and the closure doesn't seem needed given we're using classes anyways.

Reply:
huh, old habits die hard I suppose!